javascript/ql/lib/semmle/javascript/linters/ESLint.qll

```diff
@@ -62,7 +64,7 @@ module ESLint {
   /** An `.eslintrc.yaml` file. */
   private class EslintrcYaml extends Configuration instanceof YamlMapping, YamlDocument {
     EslintrcYaml() {
-      exists(string n | n = getFile().getBaseName() |
+      exists(string n | n = this.(Locatable).getFile().getBaseName() |
```

Comment:
An additional cast is required here to disambiguate the call to `getFile` due to a [difference in predicate resolution for implicit and explicit `this` receivers in constructors](https://github.com/github/codeql-core/issues/3540).